packages/central-server/src/database/createPermissionsBasedMeditrakSyncQueue.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { SqlQuery } from '@tupaia/database';
+
+const groupToArrayOrNull = field =>
+  `NULLIF(array(select distinct unnest(array_agg(${field}))), '{NULL}')`;
+
+const groupToFlatArrayOrNull = field =>
+  `NULLIF(array(select distinct unnest(array_concat_agg(${field}))), '{}')`;
+
+/**
+ * Creates the permissions_based_meditrak_sync_queue Materialized View
+ *
+ * This is used to improve the speed of querying the meditrak_sync_queue when
+ * doing a permissions based sync
+ */
+const query = new SqlQuery(`
+DROP MATERIALIZED VIEW IF EXISTS permissions_based_meditrak_sync_queue;
+CREATE MATERIALIZED VIEW permissions_based_meditrak_sync_queue AS 
+SELECT msq.*, 
```

Comment:
@edmofro bit of a change here in the latest commit. Before we had a separate field for each record type permissions, eg. (survey_countries, survey_permission_groups, survey_screen_countries, etc.)

This was pretty ugly and tedious, so I'm merging them into a single value using [COALESCE](https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-coalesce/) here. Since there's just a single value per record type I think it should all be fine, and makes things a lot more readable!